### PR TITLE
S3 bucket policies for same and cross tenant users to get object torrent

### PR DESCRIPTION
**Pull request by TejasC88**
Operation:
- create bucket user1
- Using policy s3:GetObjectTorrent, users under same and different tenants,should be able to torrent for the file

BZ exists : [[RGW]: Support for Object torrent not available](https://bugzilla.redhat.com/show_bug.cgi?id=2299815)

Fail log : http://magna002.ceph.redhat.com/cephci-jenkins/tchandra/object_torrent.txt

### CODE REVIEW

File: rgw/v2/tests/s3_swift/test_policy_torrent.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
+"""
+# test s3 bucket policies for object torrent from different users under same tenant
+
+usage : test_policy_encryption_torrent.py -c configs/<input-yaml>
+where input-yaml test_bucket_policy_delete.yaml, test_bucket_policy_modify.yaml and test_bucket_policy_replace.yaml,
+  test_bucket_policy_multiple_conflicting_statements.yaml, test_bucket_policy_multiple_statements.yaml,
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_policy_torrent.py:6]: @TejasC88 , please remove these yamls and add the actual one

**Review by hmaheswa — DISMISSED** [review]

**Review by ckulal — COMMENTED** [review]
LGTM

File: rgw/v2/tests/s3_swift/test_policy_torrent.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        bucket_name1,
+        rgw_tenant1_user1,
+        tenant1_user1_info,
+    )
+
+    if config.test_ops.get("upload_type") == "normal":
+        for oc, size in list(config.mapped_sizes.items()):
+            config.obj_size = size
+            s3_object_name = utils.gen_s3_object_name(t1_u1_bucket1.name, oc)
+            log.info("s3 object name: %s" % s3_object_name)
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_policy_torrent.py:99]: Can we use string format..

Comment (TejasC88) [reply rgw/v2/tests/s3_swift/test_policy_torrent.py]: Done , pls review

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by ckulal — APPROVED** [review]

**Pull request merged**